core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -256,4 +272,33 @@ private[spark] class AppClient(
       endpoint = null
     }
   }
+
+  /**
+   * Request executors from the Master by specifying the total number desired,
+   * including existing pending and running executors.
+   *
+   * @return whether the request is acknowledged.
+   */
+  def requestTotalExecutors(requestedTotal: Int): Boolean = {
```

Comment:
this is already done in `CoarseGrainedSchedulerBackend`. We don't need to duplicate that check.